src/nvim/main.c

```diff
@@ -2070,3 +2072,37 @@ static void check_swap_exists_action(void)
     getout(1);
   handle_swap_exists(NULL);
 }
+
+
+// Used by check_vimruntime
+static void check_vimruntime_callback(char_u *fname, void *cookie)
+{
+  STRNCPY(NameBuff, fname, MAXPATHL);
+}
+
+// Check if $VIMRUNTIME points to a valid Neovim runtime path,
+// look for $VIMRUNTIME/autoload/provider/clipboard.vim,
+// which is a Neovim-specific file, and check if it's readable.
+// Print a warning if the file is not found.
+static void check_vimruntime(void)
+{
+  int result = do_in_runtimepath((char_u*) "autoload/provider/clipboard.vim",
+    FALSE, check_vimruntime_callback, NULL);
```

Comment:
Off-topic, I just noticed that `do_in_runtimepath(NULL, TRUE, foo, NULL)` will cause a null deference.